server/s3_put_object_action.cc

```diff
@@ -623,6 +626,10 @@ void S3PutObjectAction::add_object_oid_to_probable_dead_oid_list() {
 
     // key = oldoid + "-" + newoid
     std::string old_oid_rec_key = old_oid_str + '-' + new_oid_str;
```

Comment:
For overwrite key format is <old_oid>-<new_oid>. If we expecting size based policy for both objects expected format should be

'I'<old_id>-'H'<new_oid>

Currently, it's resulting in 
'I'<old_oid>-<new_oid>